server/instrumentation-backend/src/sh/calaba/instrumentationbackend/query/ast/UIQueryASTWith.java

```diff
@@ -259,22 +286,19 @@ private boolean isMarked(Object o, Object expectedValue) {
 			return true;
 		}
 
-		ArrayList<String> getTextMethods = new ArrayList<String>();
-		getTextMethods.add("getText");
-		getTextMethods.add("getHint");
-
-		for (String methodName : getTextMethods) {
+		List<Method> methods = textMethodsForClass.get(view.getClass());
+		for (Method	method : methods) {
 			try {
-				Method getTextM = view.getClass().getMethod(methodName);
-				Object text = getTextM.invoke(view);
+				Object text = method.invoke(view);
 				if (text != null && text.toString().equals(expected)) {
 					return true;
 				}
-			} catch (Exception e) {
```

Comment:
Was there a reason for no longer doing a blanket `Exception` catch here? (Apart from the fact that it's better practice to catch specific Exceptions 🙂 )

Reply:
The two exceptions I catch are expected exceptions, anything else I would define as an coding error and I do not silently mute that.